src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java

```diff
@@ -110,55 +104,61 @@ private String normalize(String key) {
         return new String(b);
     }
 
+    @Override
     public int size() {return map.size();}
 
+    @Override
     public boolean isEmpty() {return map.isEmpty();}
 
+    @Override
     public boolean containsKey(Object key) {
-        if (key == null) {
-            return false;
-        }
-        if (!(key instanceof String)) {
-            return false;
-        }
-        return map.containsKey(normalize((String)key));
+        Objects.requireNonNull(key);
```

Comment:
I haven't looked at the CSR yet - but maybe the fact that the get and contains* method can now throw NPE - in conformance to the Map interface specification for maps that do not allow null keys or null values, should be explicitly mentioned - in particular in the release note - as this will probably be the less expected behavior change.

Reply:
Updated CSR to better reflect this behavioural change.